core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -203,6 +205,30 @@ public Result visit(Project e) {
     return builder.result();
   }
 
+  /** @see #dispatch */
+  public Result visit(Window e) {
+    Result x = visitChild(0, e.getInput());
+    Builder builder = x.builder(e);
```

Comment:
Do we need to take care of the input for window input ? `x.builder(e)` seems only work for project input.

Reply:
I did this because that the Window does not requires Clause,  it rewrites the selectlist. And it seems that x.builder(e) produces a select for Window's input node.